jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/CatalogResource.java

```diff
@@ -166,6 +185,25 @@ public Response getCatalogJson(@QueryParam(QUERY_REQUESTED_DT) final String requ
         return Response.status(Status.OK).entity(result).build();
     }
 
+    @TimedResource
+    @GET
+    @Path("/versions")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Retrieve a list of catalog versions", response = DateTime.class)
+    @ApiResponses(value = {})
+    public Response getCatalogVersionJson(@javax.ws.rs.core.Context final HttpServletRequest request) throws Exception {
+
+        final TenantContext tenantContext = context.createTenantContextNoAccountId(request);
+        final VersionedCatalog catalog = (VersionedCatalog) catalogUserApi.getCatalog(catalogName, tenantContext);
+
+        final List<DateTime> result = new ArrayList<DateTime>();
+        for (final StandaloneCatalog v : catalog.getVersions()) {
+            result.add(new DateTime(v.getEffectiveDate()));
+        }
+
+        return Response.status(Status.OK).entity(result).build();
+    }
```

Comment:
Does not this endpoint [already exist](https://github.com/killbill/killbill/blob/work-for-release-0.19.x/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/CatalogResource.java#L147) ?

Reply:
This endpoint only returns the versions, not the full json/xml.